tools/shared.py

```diff
@@ -1896,7 +1896,9 @@ def generate_string_initializer(s):
     s = s.replace('\\', '\\\\').replace("'", "\\'")
     s = s.replace('\n', '\\n').replace('\r', '\\r')
     def escape(x): return '\\x{:02x}'.format(ord(x.group()))
-    return re.sub('[\x80-\xff]', escape, s)
+    s = re.sub('\x00(?![0-9])', '\\0', s) # Don't allow a null byte in the generated output file.
+    s = re.sub('[\x00-\x1f\x80-\xff]', escape, s) # Don't allow ASCII chars 0-31 or 128-255 in the generated output file.
+    return s
```

Comment:
You might want to apply similar changes to the [uglify `to_ascii` function](https://github.com/kripken/emscripten/blob/ffb887d170000f64250c9ce4aca4e8661d677957/tools/eliminator/node_modules/uglify-js/lib/process.js#L1367-L1373).